src/murthy_lab_to_nwb/cowley2022mapping/cowley2022mappingaudiointerface.py

```diff
@@ -0,0 +1,50 @@
+"""Data interface for the audio data in this conversion."""
+from pathlib import Path
+
+from scipy.io import loadmat
+
+
+from pynwb.file import NWBFile, ProcessingModule
+from neuroconv.basedatainterface import BaseDataInterface
+from ndx_sound import AcousticWaveformSeries
+
+
+class Cowley2022MappingAudioInterface(BaseDataInterface):
+    """My behavior interface docstring"""
+
+    def __init__(self, subject: str, lobula_columnar_neuron_cell_line: str, data_dir_path: str):
+
+        audio_dir_path = Path(data_dir_path) / "raw_data" / "courtship_behavior" / "audio"
+        # 161101_10a05bin.mat
+        padded_subject = subject[3:].rjust(2, "0")
+        audio_string = f"{lobula_columnar_neuron_cell_line[2:]}{padded_subject}"
+        audio_path = next(path for path in audio_dir_path.iterdir() if audio_string in path.stem)
+        self.audio_file_path = audio_dir_path / audio_path
+        assert self.audio_file_path.is_file(), "file with audio data not found"
```

Comment:
So ultimately, this interface works off of a file path? If so I'd recommend altering the inputs to align to that. Typically only take folder path arguments if there are multiple files that have some relative structure to each other (like ordered segments/channels, etc).

Also, check out @weiglszonja's AudioInterface here: https://github.com/catalystneuro/fee-lab-to-nwb/blob/main/src/fee_lab_to_nwb/general_interfaces/audiointerface.py

It's on  her eventual to-do list to make a general one available on NeuroConv - does it seem like it might be able to work for your files? (+/- some file format extensions)

Ideally her interface would also automatically handle things like fetching sampling rate from file headers, which I think it current does for her .wav files

Reply:
Yes, I was trying to keep script folder clean (every file_path is slightly different) and passed the estimation of the exact file_path to the interfaces. Maybe all of that parsing should be done in the script. 

I looked at the interface before. Bear on mind that these are not audio files but some matlab output.

Reply:
> Bear on mind that these are not audio files but some matlab output.

Ah, OK then.

Reply:
I changed this to work in a file path fashion now.